src/storage.rs

```diff
@@ -113,11 +115,19 @@ where
         let path = Path::new(self.config.work_dir.as_ref().unwrap()); // @TODO handle unwrap explicitly
         if !path.exists() {
             debug!("creating work dir recursively: {}", path.display());
-            fs::create_dir_all(path)
+            fs::create_dir_all(path)?;
         } else {
             debug!("work dir exists: {}", path.display());
-            Ok(())
         }
+        let test_path = path.join(TEST_FILE_PATH);
```

Comment:
maybe to use `.lock` file - as mark of open directory by storage. could be helpful for debuging some case + we can store some meta there. Behavior - we can open/create this file and close on storage close.